internal/monitors/collectd/postgresql/postgresql.go

```diff
@@ -0,0 +1,184 @@
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/collectd"
+)
+
+const monitorType = "collectd/postgresql"
+
+// MONITOR(collectd/postgresql): Monitors a PostgreSQL database server using collectd's
+// [PostgreSQL plugin](https://collectd.org/wiki/index.php/Plugin:PostgreSQL).
+//
+// You have to specify each database you want to monitor individually under the
+// `databases` key.  If you have a common authentication to all databases being
+// monitored, you can specify that in the top-level `username`/`password`
+// options, otherwise they can be specified at the database level.
+//
+// Sample YAML configuration:
+//
+// ```
+// monitors:
+// - type: collectd/postgresql
+//   host: 0.0.0.0
+//   port: 5432
+//   username: "username1"
+//   password: "password1"
+//   queries:
+//   - name: "exampleQuery"
+//     pluginInstanceFrom: "testColumn"
+//     minVersion: 80203
+//     maxVersion: 80203
+//     params:
+//     - "hostname"
+//     - "database"
+//     - "instance"
+//     - "username"
+//     - "interval"
+//     statement: "Select * From test Where host = $1;"
+//     results:
+//     - type: gauge
+//       instancePrefix: testPrefix
+//       instancesFrom:
+//       - test1
+//       - test2
+//       valuesFrom:
+//       - test
+//  databases:
+//  - name: test
+//    username: "username2"
+//    password: "password2"
+//    interval: 5
+//    expireDelay: 10
+//    sslMode: disable
+//    krbSrvName: test
+//    queries:
+//    - exampleQuery
+// ```
+
+func init() {
+	monitors.Register(monitorType, func() interface{} {
+		return &Monitor{
+			*collectd.NewMonitorCore(CollectdTemplate),
+		}
+	}, &Config{})
+}
+
+// Database configures a particular PostgreSQL database
+type Database struct {
+	// the name of the database
+	Name string `yaml:"name" validate:"required"`
+	// username used to access the database
+	Username string `yaml:"username"`
+	// password used to access the database
+	Password string `yaml:"password" neverLog:"true"`
+	// interval to query the database in seconds
+	Interval int `yaml:"interval"`
+	// skip expired values in query output
+	ExpireDelay int `yaml:"expireDelay"`
+	// specify whether to use an ssl connection with PostgreSQL.
+	// (prefer(default), disable, allow, require)
+	SSLMode string `yaml:"sslMode"`
+	// specify the Kerberos service name used to authenticate with kerberos 5 or
+	// GSSAPI
+	KRBSrvName string `yaml:"krbSrvName"`
+	// queries used to generate metrics.  If no queries are specified, the
+	// default set will be used [`custom_deadlocks`, `backends`, `transactions`,
+	// `queries`, `queries_by_table`, `query_plans`, `table_states`,
+	// `query_plans_by_table`, `table_states_by_table`, `disk_io`,
+	// `disk_io_by_table`, `disk_usage`]
+	Queries []string `yaml:"queries"`
+}
+
+// Result maps values from a query to a metric
+type Result struct {
+	// type defines a metric type
```

Comment:
Titlecase the first word of all of these comments to match the style of everything else.